lib/features/sponsor/data/local/sponsor_repository_imp.dart

```diff
@@ -1,22 +1,30 @@
+import 'dart:io';
+
 import 'package:cloud_firestore/cloud_firestore.dart';
 import 'package:firebase_auth/firebase_auth.dart';
+import 'package:firebase_storage/firebase_storage.dart';
 import 'package:fiura_ecosystem/core/entities/sponsor_entity/sponsor_entity.dart';
+import '../../../images/domain/repositories/image_repository.dart';
 import '../../domain/repositories/sponsor_repository.dart';
 
 class SponsorRepositoryImp extends SponsorRepository {
   final FirebaseFirestore db;
   final FirebaseAuth auth;
+  final Reference storageRef;
 
   //Collections
   final String sponsors = "sponsors";
 
-  SponsorRepositoryImp({required this.db, required this.auth});
+  SponsorRepositoryImp(
+      {required this.db, required this.auth, required this.storageRef});
 
   @override
-  Future<bool> addSponsor(SponsorEntity sponsor) async {
+  Future<bool> addSponsor(SponsorEntity sponsor, File image) async {
     final User? user = auth.currentUser;
+    final imageRef = storageRef.child(sponsor.urlPhoto);
```

Comment:
because you do this step and the result you pass that to the Image repository.

so What Im seeing is, Reference should be inside ImageRepository and you got that step inside that repo not here